src/main.ts

```diff
@@ -13,6 +15,82 @@ async function setup(): Promise<void> {
   await setOutputs(name, url)
 }
 
+async function promote(): Promise<void> {
+  const user: string = core.getInput('user')
+  const password: string = core.getInput('password')
+  const url: string = core.getInput('url')
+  const name: string = core.getInput('name')
+  const tag: string = core.getInput('tag')
+
+  const staging_regex = new RegExp(
+    '^[^/]+:(staging|prolonged)-([0-9a-f]+).[^./]+.[0-9]+.[0-9]+$'
+  )
+  const promoted_regex = new RegExp('^[^/]+:promoted-([^/]+)$')
+
+  const staging_match = name.match(staging_regex)
+  const promoted_match = name.match(promoted_regex)
+
+  let artifacts_commit = ''
+  if (staging_match !== null) {
+    artifacts_commit = staging_match[2]
+  } else if (promoted_match !== null) {
+    artifacts_commit = promoted_match[1]
```

Comment:
The artifacts commit is not present in a promoted build, you would gather a tag here. you miss git rev-list applied on the tag